src/components/TodosContext.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Context = {
+  todos: Todo[];
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+};
+
+export const TodosContext = React.createContext<Context>({
+  todos: [],
+  setTodos: () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
```

Comment:
consider to move all states and handlers from `TodoApp` component here, inside context